server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java

```diff
@@ -468,14 +482,17 @@ private boolean shouldUseDocValuesSparseIndex(
             final IndexMode indexMode,
             final String fullFieldName
         ) {
-            if (indexed.isSet() && indexed.getValue()) {
-                return false;
-            }
-            return indexed.isConfigured() == false
-                && hasDocValues.getValue()
+            return hasDocValues.getValue()
```

Comment:
Here I remove the check on `index` value and refactored the way we check sorting on `host.name`.